Feature/Task-13 CREATED DIRECTORIES AND  DTO FOR ENTITIES

2musime opened the pull request:
## CREATING IMPORTANT DIRECTORIES AND  DTO FOR ENTITIES
This PR introduces DTOs for the Certificate, Supplier, User, and Comment entities. 

### Description
This PR introduces the Data Transfer Objects (DTOs) for the following entities in the project:
- Certificate
- Supplier
- User
- Comment
-  AssignedUser

### Changes: Files created    

- [x]     `CertificateDto`: Contains certificate details.
- [x]     `SupplierDto`: Contains supplier details.
- [x]     `UserDto`: Contains user details. 
- [x]     `CommentDto`: Contains comment details.
- [x]    `CertificateCreateDto`
- [x]    `CertificateUpdateDto`
- [x] `AssignedUserDto`

2musime committed:
Merge pull request #17 from Tumusime14/Feature/Task-12-dding-entity-classes

Feature/Task-12 Scalfolded models, entity classes were created

2musime committed:
- Created all neccessary
   directories(Controllers,Repositories,DTO,Services)
   - In DTO directory added entities dtos
   - I break down the branck into simple one.

2musime committed:
Feature/Task-13 Added CertificateCreateDto and CertificateUpdateDto
Configured .gitignore

2musime committed:
Feature/Task-13 Added CertificateCreateDto and CertificateUpdateDto

2musime committed:
Merge branch 'Feature/Task-13-adding-dto' of https://github.com/Tumusime14/Certificate-Manager---David into Feature/Task-13-adding-dto

elhasic commented on .gitignore:
```diff
@@ -253,6 +253,7 @@ src/.vs/config/applicationhost.config
-appsettings.Development.json
 
+appsettings.Development.json
+appsettings.json
```
appsettings.json should not be ignored

elhasic commented on backend/CertificateManagerAPIs/DTO/CertificateCreateDto.cs:
```diff
@@ -0,0 +1,14 @@
+        public int? UserAssigned { get; set; }
+        public int SupplierId { get; set; }
+    }
+
```
unnecessary new line

elhasic commented on backend/CertificateManagerAPIs/DTO/CertificateDto.cs:
```diff
@@ -0,0 +1,15 @@
+{
+    public class CertificateDto
+    {
+
```
unnecessary new line

elhasic commented on backend/CertificateManagerAPIs/DTO/CertificateDto.cs:
```diff
@@ -0,0 +1,15 @@
+    {
+
+        public SupplierDto Supplier { get; set; } = null!;
+        public string Type { get; set; } = null!;
```
you have to decide will you have new line between properties or not, you can not mix

elhasic commented on backend/CertificateManagerAPIs/DTO/CertificateDto.cs:
```diff
@@ -0,0 +1,15 @@
+        public DateTime ValidTo { get; set; }
+        public int Id { get; set; }
+    }
+
```
unnecessary new line

elhasic commented on backend/CertificateManagerAPIs/DTO/CertificateUpdateDto.cs:
```diff
@@ -0,0 +1,13 @@
+        public int? UserAssigned { get; set; }
+        public int SupplierId { get; set; }
+    }
+
```
unnecessary new line

elhasic commented on backend/CertificateManagerAPIs/Scripts/202409251028_certificatedb_create_tables.sql:
you cannot repeat scripts in each of this files, each file have to have only scripts that is not executed before in order how should be executed

2musime replied:
I did get this well, You mean that I `Scripts` are to be run only once, or is it to be run on each branch?
Please help me to know that.
Thank you.

2musime committed:
- Corrected unnineccessary spaces
 - Created new domain class AssignedUser
 - Added DTO for AssignedUser
 - Fixed all asked reviews.
